libraries/stdlib/src/kotlin/util/Tuples.kt

```diff
@@ -48,6 +52,30 @@ public infix fun <A, B> A.to(that: B): Pair<A, B> = Pair(this, that)
  */
 public fun <T> Pair<T, T>.toList(): List<T> = listOf(first, second)
 
+/**
+ * Transforms the first component of a [Pair] by applying the given [transform]
+ * function.
+ *  @sample samples.misc.Tuples.pairMapFirst
+ */
+public fun <A, B, T> Pair<A, B>.mapFirst(transform: (A) -> T): Pair<T, B> {
+    contract {
+        callsInPlace(transform, InvocationKind.AT_MOST_ONCE)
```

Comment:
The `transform` function is always invoked `EXACTLY_ONCE`.

Reply:
> The `transform` function is always invoked `EXACTLY_ONCE`.

Good catch, will fix that.